controllers/grid/navigationMenus/NavigationMenuItemsGridHandler.inc.php

```diff
@@ -0,0 +1,344 @@
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2003-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class NavigationMenusGridHandler
+ * @ingroup controllers_grid_navigationMenus
+ *
+ * @brief Handle navigationMenus grid requests.
+ */
+
+import('lib.pkp.classes.controllers.grid.GridHandler');
+import('lib.pkp.classes.controllers.grid.DataObjectGridCellProvider');
+import('lib.pkp.controllers.grid.navigationMenus.form.NavigationMenuItemsForm');
+import('lib.pkp.controllers.grid.navigationMenus.form.NavigationMenuItemAssignmentsForm');
+
+class NavigationMenuItemsGridHandler extends GridHandler {
+
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(
+			ROLE_ID_MANAGER,
+			array(
+				'fetchGrid', 'fetchRow',
+				'addNavigationMenuItem', 'editNavigationMenuItem',
+				'updateNavigationMenuItem',
+				'deleteNavigationMenuItem', 'saveSequence',
+				'editNavigationMenuItemAssignment', 'updateNavigationMenuItemAssignment',
+			)
+		);
+	}
+
+	//
+	// Overridden template methods
+	//
+	/**
+	 * @copydoc GridHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		//import('lib.pkp.classes.security.authorization.ContextAccessPolicy');
+		//$this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+		$context = $request->getContext();
+
+		$contextId = CONTEXT_ID_NONE;
+		if ($context) {
+			$contextId = $context->getId();
+		}
+
+		$navigationMenuItemId = $request->getUserVar('navigationMenuItemId');
+		if ($navigationMenuItemId) {
+			$navigationMenuItemDao = DAORegistry::getDAO('NavigationMenuItemDAO');
+			$navigationMenuItem = $navigationMenuItemDao->getById($navigationMenuItemId);
+			if (!$navigationMenuItem ||  $navigationMenuItem->getContextId() != $contextId) {
+				return false;
+			}
+		}
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * @copydoc GridHandler::initialize()
+	 */
+	function initialize($request, $args = null) {
+		parent::initialize($request, $args);
+
+		// Basic grid configuration
+		$this->setTitle('manager.navigationMenuItems');
+
+		// Set the no items row text
+		$this->setEmptyRowText('grid.navigationMenus.navigationMenuItems.noneExist');
+
+		// Columns
+		import('lib.pkp.controllers.grid.navigationMenus.NavigationMenuItemsGridCellProvider');
+		$navigationMenuItemsCellProvider = new NavigationMenuItemsGridCellProvider();
+		$this->addColumn(
+			new GridColumn('title',
+				'common.title',
+				null,
+				null,
+				$navigationMenuItemsCellProvider
+			)
+		);
+
+		$this->addColumn(
+			new GridColumn('path',
+				'grid.navigationMenu.navigationMenuItemPath',
+				null,
+				null,
+				$navigationMenuItemsCellProvider
+			)
+		);
+
+		$this->addColumn(
+			new GridColumn('default',
+				'common.default',
+				null,
+				null,
+				$navigationMenuItemsCellProvider
+			)
+		);
```

Comment:
I don't think the `path` or `default` columns are necessary, as they are mostly blank. Let's remove them.

Reply:
OK